### ISSUE TITLE
Unable to connect to local homeassistant address

### ISSUE THREAD

**Issue by travipross** [Jerrkawz/hassgalaxy#3]
My ISP-provided router does not support NAT Loopback ("hairpinning"), so when I'm connected to my local network, I can't access my homeassistant instance at it's external-facing address (`https://example.ddns.net`), but only at it's local address (`192.168.2.xxx`).

I've tested the normal external address in the watch app, and it works when I'm outside my home, which is great. When I'm inside my home, however, I need the address to be `192.168.2.xxx`. I'm happy enough to exclusively set it to the local address, but when I do, the watch-app never connects. 

Is there a workaround for this?

**Comment by lewtopia** [comment]
Install dnsmasq or adblock to HA through Supervisor. Use that as your DNS, and add localhost rewrite to send your domain queries to your local LAN.

**Comment by travipross** [comment]
Thank you so much for the idea! That worked (`dnsmasq` with a single `hosts` definition). Unfortunately my ISP-supplied router didn't seem to respect the manual DNS addresses provided when I configured it to point at the raspberry Pi, so I wasn't able to make all devices on my home network use this DNS rule. However, I was able to manually set DNS on my phone's connection settings for my home network, which is apparently good enough for the watch app to work while I'm in my home.

Thanks again.
